mitxgraders/comparers/comparers.py

```diff
@@ -112,30 +121,118 @@ class EqualityComparer(Comparer):
-        if self.config['transform'] is not None:
-            expected_input = self.config['transform'](expected_input)
-            student_eval = self.config['transform'](student_eval)
+    def __call__(self, comparer_params_eval, student_eval, utils):
+        expected_eval = comparer_params_eval[0]
+        self.validate(expected_eval, student_eval, utils)
 
-        return utils.within_tolerance(expected_input, student_eval)
+        transform = self.config['transform']
+        expected_eval = transform(expected_eval)
+        student_eval = transform(student_eval)
+
+        return utils.within_tolerance(expected_eval, student_eval)
 
 equality_comparer = EqualityComparer()
 
+class MatrixEntryComparer(CorrelatedComparer):
+    """
+    Default comparer for MatrixGrader. Compares student and instructor matrix
+    evaluations entry-by-entry for equality.
+
+    Configuration
+    =============
+
+        transform (None | function): same as EqualityComparer (default None)
+        entry_partial_credit ('proportional' | number): Determines how partial credit
+            is awarded. If set to 'proportional', then credit is proportional to
+            the number of correct matrix entries. If a numeric value betweem 0 and 1
+            is provided, this flat rate of partial credit is provided as long as
+            some but not all entries are correct. Default is the numeric value 0
+            (no partial credit).
+
+        entry_partial_msg (str): A text string message shown when partial credit
+            is awarded. The string may optionally contain the formatting key {error_indices},
+            which will be replaced with the indices of the incorrect matrix entries.
+            Default value is:
+            "Some matrix entries are incorrect, marked below:\n{error_locations}"
+    """
+
+    default_msg = "Some matrix entries are incorrect, marked below:\n{error_locations}"
+    schema_config = EqualityComparer.schema_config.extend({
+        Required('entry_partial_credit', default=0): Any(All(Number, Range(0, 1)), 'proportional'),
+        Required('entry_partial_msg', default=default_msg): text_string
+    })
+
+    @staticmethod
+    def format_message_with_locations(format_string, locs):
+        """
+        Returns format_string with {error_locations} replaced by 1-indexed error
+        locations.
+
+        Arguments:
+            format_string: a string that may contain {error_locations} formatting key.
+            locs: a boolean error with False values indicating errors
+        """
+        # Not the most elegant way to do these replacements, but this was what
+        # I came up with the minimize the amount of extra u prefixes in Python 2
+        bad_str = '\u2717'
+        good_str = '\u2713'
+        matrix_as_text = six.text_type(locs).replace("  ", " ").replace("[ ", "[")
+        matrix_as_text = matrix_as_text.replace("True", good_str).replace("False", bad_str)
```

Comment:
I might wrap this in a `<code></code>` block to make sure that spacing looks ok. I'll test and get back to you.

Reply:
So, I tested this out on edX. Code blocks don't work so well, but pre blocks do the job beautifully:
```html
<pre>[[✓ <span style='color:red;'>✗</span>]<br/> [<span style='color:red;'>✗</span> ✓]]</pre>
```
I added in the colors (just update your bad_str appropriately.

One thing to note is that the `<pre>` blocks need to have `\n` replaced with `<br/>` but not `<br/>\n` as is done by default, so you'd need to do the replacement here rather than rely on the default replacements we perform.

Will also require updating tests minimally.